### PR TITLE
fix: check if func binaries exist before starting

### LINKED ISSUE #38: Check if Azure Functions Core Tools is available before starting the api
**Is your feature request related to a problem? Please describe.**
When the swa cli tries to star the api project using the `func` cli (Azure Functions Core Tools). If `func` is not available, this would trigger an error, an the api won't be emulated.

**Describe the solution you'd like**
Check if the `func` cli is already installed. If not, show a warning message or abort the whole process.

**Additional context**
This should work across all supported platforms.

cc @anthonychu

### PR DESCRIPTION
**Pull request by manekinekko**
Closes #38

### CODE REVIEW

**Comment by manekinekko** [comment]
@aaronpowell would you be able to validate this PR on a Windows machine? 🙏

**Comment by aaronpowell** [comment]
WSL count 😝

I'll do it today

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: src/utils.ts [hunk 1]
```diff
@@ -380,18 +380,31 @@ export function computeAppLocationFromArtifactLocation(appArtifactLocation: stri
...
   if (binDirErr) {
-    console.error({ binDirErr });
+    console.error(`ERRR: `, binDirErr);
+    process.exit(-1);
+  }
+  return binDirOutput.stdout.toString().trim();
+}
+
+export function getBinaryPath(binary: string) {
+  const binDirOut = exec(isWindows() ? "npm.cmd" : "npm", ["bin"]);
```
Comment (aaronpowell) [inline src/utils.ts:394]: You don't need to search for `npm.cmd` on Windows, assuming that `npm` is correctly in the path it will take care of it for you

**Review by aaronpowell — CHANGES_REQUESTED** [review]
Trying to traverse the file system across different OS/shell combinations is going to be fraught with issues (for example, the current Windows detection won't work in PowerShell).

Is there a reason why you wouldn't just use `npx` to execute the commands? That's already got the file system traversal configured so you don't need to do it yourself, and it'll fall back to installing the missing tools as well.

**Comment by manekinekko** [comment]
Thank you @aaronpowell.

For `npx`, I've tried different combinations of options, but each time, `npx` was causing some issues:
- it tried to always download `concurrently` and `http-server` (adding 10+ seconds of starting time when running `swa start ...`).
- curiously, when using flags with `npx --quite`, was causing [these lines](https://github.com/Azure/static-web-apps-cli/blob/main/src/cli/commands/start.ts#L115-L118) to break!

**Comment by aaronpowell** [comment]
weird, with #24 I didn't experience any of those problems. Did you try `--no-install` on `npx` to see what will happen?

**Comment by manekinekko** [comment]
Yes, it still had to install the deps!  Would you mind giving it another try and report your results?

`npx --quite --no-install http-server`

**Comment by aaronpowell** [comment]
_sigh_

Looks like on Windows we have a bigger problem, `npx` has a bug in handling command line arguments - https://github.com/npm/npx/issues/43

I think a better bet would be to programmatically use concurrently, rather than running it from `spawn`.

**Comment by aaronpowell** [comment]
I've opened #71 which shows using `concurrently` programmatically and it can start `http-server` on Windows successfully. There's still some work to be done to get the api working, but I'll wait for a review of #71.

**Comment by manekinekko** [comment]
Closing in favor of #71

**Pull request closed without merging**